applications/acdc/src/acdc_agent_listener.erl

```diff
@@ -86,7 +86,7 @@
          ,is_thief = 'false' :: boolean()
          ,agent :: agent()
          ,agent_call_ids = [] :: api_binaries() | wh_proplist()
-         ,cdr_urls = dict:new() :: dict() %% {CallId, Url}
+         ,cdr_urls = dict:new() :: dict:dict() %% {CallId, Url}
```

Comment:
There's `nowarn_deprecated_type` over at http://www.erlang.org/doc/man/compile.html FYI.

To really switch to 17 there's a bunch of gotchas that will happen at runtime (or maybe those can get caught running xref/dialyzer using 17). One way to find most of them is to go through all CHANGELOGs in between releases.
I know there'll be some issues with the sha1 and md5 functions from the crypto module…